src/optimizer_interface.jl

```diff
@@ -720,6 +720,13 @@ function MOI.get(
     return _moi_get_result(backend(model), attr, index(v))
 end
 
+function get_variable_attribute(
```

Comment:
I think we either need `get_variable_attribute(model, attr, x)` or `get_variable_attribute(x, attr)`.

There's also an argument that we should just overload `get_optimizer_attribute(x, attr)` and  `set_optimizer_attribute(x, attr, value)`.

Reply:
> I think we either need get_variable_attribute(model, attr, x) or get_variable_attribute(x, attr).

omitting the model was suggested in: https://github.com/jump-dev/ParametricOptInterface.jl/issues/121#issuecomment-1430395250

will switch to `get_variable_attribute(x, attr)`

> There's also an argument that we should just overload get_optimizer_attribute(x, attr) and set_optimizer_attribute(x, attr, value).

I understand, I am not sure which is better. Totaly to open to change,
Since we have: `MOI.AbstractOptimizerAttribute` we could reserve `get_optimizer_attribute` for it.
And  `MOI.AbstractModelAttribute` would match with `get_model_attribute`.

Reply:
I don't think users really care about the distinction of optimizer/model/variable/constraint. 

At the JuMP level, `get_optimizer_attribute` could just mean "query something from the bottom-most solver"

Reply:
I'd be fine with that.

Reply:
maybe `get_attribute` for all? since `get_optimizer_attribute(x, attr)` without the model parameters might be slightly weird?

Reply:
This feels like a naming question for @mlubin

Reply:
I prefer not adding in the name something that is in the signature so I would prefer `get_attribute` and `set_attribute`

Reply:
Updated to use `get/set_attribute`. I think the `getindex` feels a bit too cute? `model[MOI.Silent()] = true`? `x[MOI.VariableName]`? I guess that makes sense, but it doesn't feel very JuMP-like. (More Pyomo-like.)

Reply:
I like `get/set_attribute`. It is very consistent with everything in JuMP.

>  I think the getindex feels a bit too cute? model[MOI.Silent()] = true? x[MOI.VariableName]? I guess that makes sense, but it doesn't feel very JuMP-like. (More Pyomo-like.)

I agree and also have mixed feelings. We can consider that and bikeshed, I would just leave it for another PR/issue and merge this one as it seems a nice improvement.

Reply:
We already have that for getting your variables from they name with `model[:x]`. We're not aiming for cute syntax at the MOI level but we can't say that it's not the purpose of JuMP. `getindex`, `setindex!` is a good way to have a nice and consistent syntax for these kind of things.

Reply:
I guess there's a difference between things that JuMP/the model owns, like `:x`, and attributes like `MOI.Silent`. Regardless, can we leave this for a later discussion? See how people use this first?

Reply:
But then we might get the same issue that `get_optimizer_attribute` that we now want to remove but it's breaking.
I guess here we could support both syntax

Reply:
> But then we might get the same issue

I don't think we're ever going to want people to set options using `model["output_flag"] = "off"`?